libs/ledgerjs/packages/hw-app-eth/src/modules/EIP712/index.ts

```diff
@@ -67,12 +79,23 @@ const makeRecursiveFieldStructImplem = (
         if (fieldType) {
```

Comment:
I think it shouldn't be the case but if fieldName mandatory here ?

Reply:
Hmm good question... I don't know what's best here because the format of an EIP712 is this:
```javascript
{
  domain: {
    name: "Whatever",
  },
  types: {
    EIP712Domain: [
      {
        name: "name",
        type: "string",
      },
    ],
    hello: [
      { name: "foo", type: "string" }
    ],
  },
  primaryType: "hello",
  message: {
    hello: {
      foo: "bar",
    },
  },
};
```

and `typesMap` is composed with the types in the `types` key, while `fieldName` is coming from `message` key, which means for a broken EIP712 message `fieldType` could theorically be undefined. 
Then I don't know if we should throw an error as the JSON isn't compliant with the standard, or just skip it like I did ? 🤔